KIPs/kip-xx.md

```diff
@@ -0,0 +1,57 @@
+---
+kip: to be assigned
+title: Transaction data gas cost reduction
+author: Lake (@hyunsooda) and Ollie (@blukat29)
+discussions-to: https://github.com/kaiachain/kips/issues/22
+status: Draft
+type: Standards Track
+category: Core
+created: 2024-10-18
+---
+
+## Abstract
+
+Reduce the calldata gas costs to the [EIP-2028](https://eips.ethereum.org/EIPS/eip-2028) level. The purpose is to achieve better compatibility with Ethereum-oriented SDKs and tools.
+
+## Motivation
+
+Since the introduction of [EIP-2028](https://eips.ethereum.org/EIPS/eip-2028), several bundler software solutions have been developed.
+
+The `eth_estimateUserOperationGas` API calculates the gas required for a userOperation. This estimate is computed based on the input’s composition, which includes both nonzero and zero bytes. Following the Istanbul hardfork, Ethereum set the gas cost at 16 gas per nonzero byte and 4 gas per zero byte.
+As a result, some bundlers have adopted predefined gas costs for these byte types.
+
+To maintain compatibility with Ethereum's tooling ecosystem, the Kaia mainnet aligns its gas pricing with the specifications of EIP-2028 introduced in the Prague hardfork, which defines new costs for nonzero and zero bytes.
+
+## Specification
+
+The gas cost distinguishes between non-zero and zero bytes, assigning 16 and 4 gas units to each, respectively.
```

Comment:
The gas costs for calldata is calculated from the number of nonzero and zero bytes, incurring 16 and 4 gas units to each.

Reply:
Refined.